### ISSUE TITLE
Build without geos is broken

### ISSUE THREAD

**Issue by dcesari** [ARPA-SIMC/arkimet#123]
A build without geos library fails:
```
./configure --disable-static --enable-bufr --disable-vm2 --disable-geos --disable-python --prefix=/scratch/dcesari/smnd
...
make
...
libarkimet_la-bbox.lo `test -f 'bbox.cc' || echo './'`bbox.cc
libtool: compile:  g++ -DHAVE_CONFIG_H -I.. -I.. -L/usr/lib -DCONF_DIR=\"/scratch/dcesari/smnd/etc/arkimet\" -DPOSTPROC_DIR=\"/scratch/dcesari/smnd/lib/arkimet\" -I/scratch/dcesari/smnd/include -Werror -I/scratch/dcesari/smnd/include -I/scratch/dcesari/smnd/include/dballe -Wall -g -O2 -MT libarkimet_la-bbox.lo -MD -MP -MF .deps/libarkimet_la-bbox.Tpo -c bbox.cc  -fPIC -DPIC -o .libs/libarkimet_la-bbox.o
In file included from ../arki/types/area.h:7:0,
                 from ../arki/bbox.h:4,
                 from bbox.cc:2:
bbox.cc: In constructor ‘arki::BBox::BBox(const string&)’:
../arki/utils/geosfwd.h:60:22: error: ‘getDefaultInstance’ is not a member of ‘dummygeos::DummyGeos’
 #define ARKI_GEOS_NS dummygeos
                      ^
../arki/utils/geosfwd.h:62:35: note: in expansion of macro ‘ARKI_GEOS_NS’
 #define ARKI_GEOS_GEOMETRYFACTORY ARKI_GEOS_NS::DummyGeos
                                   ^~~~~~~~~~~~
bbox.cc:66:54: note: in expansion of macro ‘ARKI_GEOS_GEOMETRYFACTORY’
 BBox::BBox(const std::string& code) : L(new Lua), gf(ARKI_GEOS_GEOMETRYFACTORY::getDefaultInstance()), funcCount(0)
                                                      ^~~~~~~~~~~~~~~~~~~~~~~~~
Makefile:2919: set di istruzioni per l'obiettivo "libarkimet_la-bbox.lo" non riuscito
```
If it's hard to maintain, better remove the chance to disable geos.

**Comment by spanezz** [comment]
It's not hard to maintain, it's just not usually tested. The ideal situation would be to have a CI system that regularly performs rebuilds+tests with various enable/disable combinations

**Comment by spanezz** [comment]
Fixed in d40638b011d42d7a12d9c6ccd95358fd3af0fc4d

**Comment by spanezz** [comment]
One test still fails without libvm2, because it runs a Lua test code that assumes VM2 support is there, and currently I cannot think of a way to test for existance of VM2 support from Lua
